packages/flutter/lib/src/gestures/events.dart

```diff
@@ -1731,6 +1731,22 @@ abstract class PointerSignalEvent extends PointerEvent {
   });
 }
 
+/// A function used to send information about a PointerEvent back to the engine.
+///
+/// This is used on the web to `preventDefault`.
+typedef RespondPointerEventFn = void Function({required bool allowPlatformDefault});
+
+mixin _RespondablePointerEvent on PointerEvent {
```

Comment:
Just wondering why this is implemented as a mixin instead of just defining the respond method directly on PointerSignalEvent?

Reply:
This is implemented as a mixin, because the "_Transformed" subclasses of `PointerSignalEvent` are `implements` (not extends), so I'd have to duplicate the empty method [a](https://github.com/flutter/flutter/blob/bc097c619af17c9c66531a085b9d99874a944c5f/packages/flutter/lib/src/gestures/events.dart#L1817) [few](https://github.com/flutter/flutter/blob/bc097c619af17c9c66531a085b9d99874a944c5f/packages/flutter/lib/src/gestures/events.dart#L1908) [times](https://github.com/flutter/flutter/blob/bc097c619af17c9c66531a085b9d99874a944c5f/packages/flutter/lib/src/gestures/events.dart#L1999). (See the `_Transformed` classes of the `extends PointerSignalEvent` events).

(PS: Not a fan of the OOP design of this file, it's starting to show its age!)

(PPS: If there's a cleaner way of doing this, I'm all ears. I just didn't want to add more cruft to the base class, having a specific `PointerSignalEvent` class.